app/main.py

```diff
@@ -97,37 +97,57 @@ def grab_secrets(self):
 
         if secrets_keys is not None:
             for key_info in filter(None, secrets_keys.split(';')):
-                key_name, _, key_version = key_info.strip().partition(':')
-                _logger.info('Retrieving secret name:%s with version: %s', key_name, key_version)
+                key_name, key_version, cert_filename, key_filename = self._split_keyinfo(key_info)
+                _logger.info('Retrieving cert name:%s with version: %s output certFileName: %s keyFileName: %s', key_name, key_version, cert_filename, key_filename)
                 secret = client.get_secret(vault_base_url, key_name, key_version)
                 output_path = os.path.join(self._secrets_output_folder, key_name)
                 if secret.kid is not None:
                     _logger.info('Secret is backing certificate. Dumping private key and certificate.')
-                    self._dump_pfx(secret.value, key_name)
+                    self._dump_pfx(secret.value, cert_filename, key_filename)
                 _logger.info('Dumping secret value to: %s', output_path)
                 with open(output_path, 'w') as secret_file:
                     secret_file.write(secret.value)
 
         if certs_keys is not None:
             for key_info in filter(None, certs_keys.split(';')):
-                key_name, _, key_version = key_info.strip().partition(':')
-                _logger.info('Retrieving cert name:%s with version: %s', key_name, key_version)
+                key_name, key_version, cert_filename, key_filename = self._split_keyinfo(key_info)
```

Comment:
You should put a comment here that you really only care about the cert_filename.  Actually, I would make key_filename be just "_" here such that it is clear you don't care about that part of the result.

Reply:
Yes will make that change